tests/test_measure_companions.py

```diff
@@ -0,0 +1,156 @@
+
+    Args:
+        out_dir (str): Output directory for saved FITS files.
+
+    Returns:
+        direct_star_image (corgidrp.data.Image):
+        host_star_counts (float):
+        zero_point (float):
+        ct_cal (corgidrp.data.CoreThroughputCalibration):
+        FpamFsamCal (corgidrp.data.FpamFsamCal):
+        psf_sub_frame (corgidrp.data.Image):
+    """
+    os.makedirs(out_dir, exist_ok=True)
+
+    # Generate Direct Star Image
+    star_flux = nd_filter_calibration.compute_expected_band_irradiance(INPUT_STARS[0], "3C")
+    direct_star_image = mocks.create_flux_image(
+        star_flux, FWHM, CAL_FACTOR, "3C", "HOLE", INPUT_STARS[0],
+        fsm_x=0, fsm_y=0, exptime=5, filedir=out_dir,
+        color_cor=1.0, platescale=21.8, background=0,
+        add_gauss_noise=False, noise_scale=1.0, file_save=True
+    )
+
+    # Compute Host Star Counts
+    host_star_counts, _, _ = fluxcal.aper_phot(
+        direct_star_image, encircled_radius=10, frac_enc_energy=1.0, method='subpixel',
+        subpixels=5, background_sub=True, r_in=12, r_out=20, centering_method='xy',
+        centroid_roi_radius=10
+    )
+
+    print(f"\nMeasured host star counts: {host_star_counts:.5f} photoelectrons")
+
+    # Compute Zero Point
+    fluxcal_factor = get_fluxcal_factor(direct_star_image, PHOT_METHOD, PHOT_ARGS, FLUX_OR_IRR)
+    zero_point = fluxcal_factor.ext_hdr.get('ZP', None)
+
+    # Generate Calibration Data
+    dataset_ct, ct_cal = mocks.create_mock_ct_dataset_and_cal_file(fwhm=50, n_psfs=20, cfam_name='3C', save_cal_file=True)
+    FpamFsamCal = mocks.create_mock_fpamfsam_cal(save_file=False)
+
+    print("Companion 1 AP Mag:", (-2.5 * np.log10(host_star_counts / 2) + zero_point))
+    print("Companion 2 AP Mag:", (-2.5 * np.log10(host_star_counts / 3) + zero_point))
+
+    # Generate PSF-subtracted Frame
+    psf_sub_frame = mocks.generate_psfsub_image_with_companions(
+        nx=200, ny=200, host_star_center=None, host_star_counts=host_star_counts,
+        psf_sub_scale=0.7, companion_xy=[(120, 80), (90, 130)],
+        companion_counts=[host_star_counts / 2, host_star_counts / 3],
+        companion_mags=[(-2.5 * np.log10(host_star_counts / 2) + zero_point),
+                        (-2.5 * np.log10(host_star_counts / 3) + zero_point)],
+        zero_point=zero_point, ct_cal=ct_cal, use_ct_cal=True, cor_dataset=dataset_ct,
+        FpamFsamCal=FpamFsamCal, blur_sigma=0.5, noise_std=1e-8, outdir=out_dir, platescale=0.0218
+    )
+
+    coron_data = mocks.generate_coron_dataset_with_companions(
+        n_frames=1, shape=(200, 200), companion_xy=[(120, 80), (90, 130)], companion_counts=[host_star_counts / 2, host_star_counts / 3],
+        host_star_counts=host_star_counts, roll_angles=[10.0], add_noise=True, noise_std=1e-8, outdir=out_dir
+    )
+
+    return direct_star_image, host_star_counts, fluxcal_factor, zero_point, dataset_ct, ct_cal, FpamFsamCal, psf_sub_frame, coron_data
+
+
+def test_measure_companions_wcs(out_dir):
```

Comment:
pytest tests shouldn't have unspecified args, so we should default to a test dir